src/ide/code_lens/mod.rs

```diff
@@ -18,17 +22,50 @@ pub struct CodeLensControllerState {
     lens: HashMap<Url, Vec<(CodeLens, CodeLensKind)>>,
 }
 
-#[derive(Clone, Default)]
+#[derive(Clone)]
 pub struct CodeLensController {
     state: Arc<RwLock<CodeLensControllerState>>,
+    refresh_sender: Sender<RefreshCodeLensRequest>,
+    request_refresh_receiver: Receiver<()>,
+    // Keep it last so we can drop channels.
+    // Otherwise, the refresh thread will never stop, and the it's
+    // JoinHandle drop will cause deadlock by waiting for the thread to join.
+    _refresh_thread: Arc<JoinHandle<()>>,
 }
 
 impl CodeLensController {
-    pub fn refresh_all_lenses(
+    pub fn new() -> Self {
+        let (refresh_sender, refresh_receiver) = channel::unbounded();
+        let (request_refresh_sender, request_refresh_receiver) = channel::bounded(1);
```

Comment:
Imho would be good to add a comment explaining why those 2 channels got those certain limits. Just briefly explain